src/System.Threading.Channels/src/System/Threading/Channels/BoundedChannel.cs

```diff
@@ -168,6 +168,11 @@ public override bool TryComplete(Exception error)
 
                     // Mark that we're done writing.
                     parent._doneWriting = error ?? ChannelUtilities.s_doneWritingSentinel;
+
+                    ChannelUtilities.FailInteractors<WriterInteractor<T>, VoidResult>(parent._blockedWriters, ChannelUtilities.CreateInvalidCompletionException(error));
+                    ChannelUtilities.WakeUpWaiters(ref parent._waitingReaders, result: false, error: error);
+                    ChannelUtilities.WakeUpWaiters(ref parent._waitingWriters, result: false, error: error);
```

Comment:
I agree there's an issue to be fixed here, but the fix is broken. If AllowSynchronousContinuations is true and there are any waiting readers, these readers will be invoked synchronously while holding the lock, which we must avoid.

Reply:
will take a look and try to figure out what is the best fix